doc/examples/realtime_servo/realtime_servo_tutorial.rst

```diff
@@ -1,181 +1,196 @@
+    // Create the command.
+    JointJogCommand command;
+    command.joint_names = {"panda_link7"};
+    command.velocities = {0.1};
 
-Introspection
--------------
+    // Set JointJogCommand as the input type.
+    servo.setCommandType(CommandType::JOINT_JOG);
 
-Here are some tips for inspecting and/or debugging the system.
+    // Get the joint states required to follow the command.
+    // This is generally run in a loop.
+    KinematicState next_joint_state = servo.getNextJointState(command);
 
-#. View the :code:`ros2_controllers` that are currently active with :code:`ros2 control list_controllers`. You will see a `JointTrajectoryController <https://github.com/ros-controls/ros2_controllers/tree/master/joint_trajectory_controller>`_ that receives the joint position commands from Servo and handles them in the simulated robot driver. The JointTrajectoryController is very flexible; it can handle any combination of position/velocity/(position-and-velocity) input. Servo is also compatible with `JointGroupPosition <https://github.com/ros-controls/ros2_controllers/tree/master/position_controllers>`_ or `JointGroupVelocity <https://github.com/ros-controls/ros2_controllers/tree/master/velocity_controllers>`_-type controllers.
+Using the TwistCommand
 
-#. :code:`ros2 topic echo /servo_node/status` shows the current state of the Servo node. If :code:`0` is published, all is well. The definition for all enums can be seen :moveit_codedir:`here.<moveit_ros/moveit_servo/include/moveit_servo/status_codes.h>`
+.. code-block:: c++
 
-#. :code:`ros2 node list` shows the following. :code:`ros2 node info` can be used to get more information about any of these nodes.
+    using namespace moveit_servo;
 
-   - :code:`/joy_node` handles commands from the XBox controller
+    // Create the command.
+    TwistCommand command{"panda_link0", {0.1, 0.0, 0.0, 0.0, 0.0, 0.0};
 
-   - :code:`/moveit_servo_demo_container` holds several ancillary ROS2 "component nodes" that are placed in a container for faster intra-process communication
+    // Set the command type.
+    servo.setCommandType(CommandType::TWIST);
 
-   - :code:`/servo_node` which does the calculations and collision checking for this demo. :code:`servo_node` may be moved into the demo container in the future
+    // Get the joint states required to follow the command.
+    // This is generally run in a loop.
+    KinematicState next_joint_state = servo.getNextJointState(command);
 
-Using the C++ Interface
------------------------
-Instead of launching Servo as its own component, you can include Servo in your own nodes via the C++ interface. Sending commands to the robot is very similar in both cases, but for the C++ interface a little bit of setup for Servo is necessary. In exchange, you will be able to directly interact with Servo through its C++ API.
 
-This basic C++ interface demo moves the robot in a predetermined way and can be launched with ::
+Using the PoseCommand
```

Comment:
Might be nice to expand a bit more on the text to officially define what `JointJogCommand`, `TwistCommand`, and `PoseCommand` actually do.

Reply:
I agree with Andy on this,  we are already linking the `datatypes.hpp` header in the beginning of the section so that users can take a look at that before proceeding to the section where we show how to use the commands.